lua/barbar/ui/render.lua

```diff
@@ -85,202 +69,12 @@ local scroll = { current = 0, target = 0 }
 --- @class barbar.ui.Render
 local render = {}
 
---- An incremental animation for `close_buffer_animated`.
```

Comment:
The open/close code has been moved back to `state`, where state management should live. I think this might have been done to avoid circular dependencies. I've solved the state/render dependency by adding `state.update_callback`, which is set to `render.update` in `render.lua`. Think of it like an event listener callback, when the state changes, the update callback is called.

Reply:
I think it is important for each module to have a single responsibility, and that support of additional responsibilities should be built on top of it rather than into it. 

In my view `state`'s job is:

* Define what the shape of the shared state is
* Provide a minimal interface to read and write state that ensures internal integrity

What do you think about splitting out the other responsibilities (e.g. `get_updated_buffers`, `xyz_animated`) into other modules?

Reply:
The functions I've moved here are too tied to the state to live elsewhere. The animations require setting fields on the state object, there's no avoiding that. I'm open to suggestions but I don't think it will be possible to split them like we do for the layout module.